watchtower-module-eslint/src/main/java/com/tracelink/appsec/module/eslint/EsLintModule.java

```diff
@@ -62,15 +67,15 @@ public String getMigrationsLocation() {
 	 */
 	@Override
 	public IScanner getScanner() {
-		return new EsLintScanner();
+		return new EsLintScanner(engine);
 	}
 
 	/**
 	 * {@inheritDoc}
 	 */
 	@Override
 	public IRuleDesigner getRuleDesigner() {
-		return new EsLintRuleDesigner();
+		return designer;
```

Comment:
The designer has to be a service because the controller needs it. As no other spring-y things need the other objects, they don't need to be handled as components.